core/src/main/java/org/kiwix/kiwixmobile/core/base/BasePresenter.kt

```diff
@@ -15,30 +15,26 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.base;
+package org.kiwix.kiwixmobile.core.base
 
-import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.CompositeDisposable
+import org.kiwix.kiwixmobile.core.base.BaseContract.Presenter
+import org.kiwix.kiwixmobile.core.base.BaseContract.View
 
 /**
  * All presenters should inherit from this presenter.
  */
-
-public abstract class BasePresenter<T extends BaseContract.View>
-  implements BaseContract.Presenter<T> {
-
-  protected final CompositeDisposable compositeDisposable = new CompositeDisposable();
-  protected T view;
-
-  @Override
-  public void attachView(T view) {
-    this.view = view;
+abstract class BasePresenter<T : View<*>?> : Presenter<T> {
+  val compositeDisposable = CompositeDisposable()
```

Comment:
to prevent changes to the other files as part of this you can use `@JvmField` and then you won't need to call the getters of the properties